### PR TITLE
[Pipeline](load) Support transaction on pipeline engine

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/runtime/fragment_mgr.cpp [hunk 1]
```diff
@@ -571,14 +579,24 @@ void FragmentMgr::set_pipe(const TUniqueId& fragment_instance_id,
...
 
-std::shared_ptr<io::StreamLoadPipe> FragmentMgr::get_pipe(const TUniqueId& fragment_instance_id) {
+std::shared_ptr<io::StreamLoadPipe> FragmentMgr::get_pipe(const TUniqueId& fragment_instance_id,
+                                                          bool enable_pipeline_engine) {
     {
         std::lock_guard<std::mutex> lock(_lock);
-        auto iter = _fragment_map.find(fragment_instance_id);
-        if (iter != _fragment_map.end()) {
-            return _fragment_map[fragment_instance_id]->get_pipe();
+        if (enable_pipeline_engine) {
```
Comment (HappenLee) [inline be/src/runtime/fragment_mgr.cpp:586]: maybe should do double search and remove useless `enable_pipeline_engine` in function param and thrift struct

Comment (HappenLee) [inline be/src/runtime/fragment_mgr.cpp:586]: maybe should do double search and remove useless `enable_pipeline_engine` in function param and thrift struct

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.04 seconds
 load time: 649 seconds
 storage size: 17122935454 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230105055315_clickbench_pr_74027.html

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**